Autonomous decision making demonstration.ipynb

```diff
@@ -208,51 +205,39 @@
     "        path.insert(0, curr)\n",
     "    return path\n",
     "\n",
-    "def edge_distance_between(node_from, node_to, grid):\n",
-    "    \"\"\"\n",
-    "    The edge distance between any two nodes is the cost of traversing node_from.\n",
-    "    That is, in this context, node_from is actually irrelevant. We'll retain\n",
-    "    the API for this function as is, however, in case our representation of the\n",
-    "    graph changes in future.\n",
-    "    \"\"\"\n",
-    "    a = grid[node_to[0]][node_to[1]]# Tuneable parameter to adjust how averse to obstacles the algorithm is.\n",
-    "    return a\n",
-    "\n",
-    "def neighbors_of(node, grid):\n",
+    "def get_successors(node, grid):\n",
     "    \"\"\"\n",
     "    The neighbors of a cell (node) in the grid are the 8-surrounding cells.\n",
     "    \"\"\"\n",
-    "    neighbors = []\n",
+    "    successors = []\n",
     "\n",
+    "    nodex, nodey = node\n",
     "    n_rows = len(grid)\n",
     "    n_cols = len(grid[0])\n",
     "\n",
-    "    for r_delta in [-1, 0, 1]:\n",
-    "        for c_delta in [-1, 0, 1]:\n",
-    "            r = node[0] + r_delta\n",
-    "            c = node[1] + c_delta\n",
+    "    for dx, dy in product([-1,0,1],[-1,0,1]):\n",
+    "        # skip the current node itself\n",
+    "        if (dx == 0 and dy == 0):\n",
+    "            continue\n",
     "\n",
-    "            # skip neighbors that would take us off the edge of the grid\n",
-    "            if r < 0 or r >= n_rows or c < 0 or c >= n_cols:\n",
-    "                continue\n",
-    "            \n",
-    "            # skip the current node itself\n",
-    "            if r_delta == 0 and c_delta == 0:\n",
-    "                continue\n",
-    "            if grid[c][r]>1.1:\n",
-    "                continue\n",
-    "        \n",
-    "            neighbors.append( (r, c) )\n",
+    "        x = nodex + dx\n",
+    "        y = nodey + dy\n",
```

Comment:
Can you change **nodey** to node_y or something - I keep reading it as Nodee. lol

Reply:
Probably do the same with node_x for sameness

Reply:
lol I'll change it in the next commit